src/ethereum/mod.rs

```diff
@@ -239,102 +220,18 @@ impl Ethereum {
     }
 }
 
-impl From<Address> for H160 {
-    /// Creates an H160 type from an address.
-    /// H160 will equal the address's bytes.
-    fn from(address: Address) -> H160 {
-        let bytes: [u8; 20] = address.into();
-        H160::from(bytes)
-    }
-}
-
-impl From<H160> for Address {
-    /// Converts an H160 type to an Address.
-    /// The address's bytes will be a copy of H160.
-    fn from(h160: H160) -> Address {
-        h160.0.into()
-    }
-}
-
-impl From<Web3Bytes> for Bytes {
-    fn from(bytes: Web3Bytes) -> Bytes {
-        bytes.0.into()
-    }
-}
-
-impl From<Web3H256> for H256 {
-    /// Converts a web3 H256 into an `H256`.
-    fn from(h256: Web3H256) -> H256 {
-        h256.0.into()
-    }
-}
-
-impl From<Web3U128> for U128 {
-    /// Converts a web3 U128 into a `U128`.
-    fn from(u128: Web3U128) -> U128 {
-        u128.0.into()
-    }
-}
-
-impl From<Web3U256> for U256 {
-    /// Converts a web3 U256 into a `U256`.
-    fn from(u256: Web3U256) -> U256 {
-        u256.0.into()
-    }
-}
-
-impl From<H520> for Signature {
-    /// Converts a web3 H520 into a `Signature`.
-    fn from(h520: H520) -> Signature {
-        Signature { 0: h520.0 }
-    }
-}
-
-impl From<U128> for BlockNumber {
-    /// Tries to convert a U128 into a block number. Panics if there is a u64 overflow during
-    /// conversion.
-    fn from(u128: U128) -> BlockNumber {
-        let block_number: u64 = u128.into();
-
-        BlockNumber::from(block_number)
-    }
-}
-
 impl From<Log> for Event {
```

Comment:
Why can we use `web3.types.Log`? why do we need `Event`  type 🤔

Reply:
`Log` only stores low-level data like hashed topic and ABI encoded data. The events are supposed to have meaningful fields and data.

Reply:
`log.rs` from `web3 types` contains similar fields like `Event` in ethereum types.  

Do you mean to deserialize `data` field in the events?  

https://github.com/tomusdrw/rust-web3/blob/master/src/types/log.rs

Reply:
Yes, deserialized `data` field and also a readable "topic".